org/mozilla/jss/JSSProvider.java

```diff
@@ -9,6 +9,7 @@
 import java.io.InputStream;
 
 public final class JSSProvider extends java.security.Provider {
+    public static boolean SSLEngine = false;
```

Comment:
The `SSLEngine` looks like a class name instead of variable name. I'd probably use something like `USE_SSL_ENGINE`. Regardless, this is just preference. The code looks good, feel free to merge.